apps/backend/src/applications/applications.service.ts

```diff
@@ -22,6 +24,26 @@ export class ApplicationsService {
     return apps;
   }
 
+  async findAllCurrentApplications(): Promise<GetApplicationResponseDTO[]> {
+    const currentCycle: Cycle = getCurrentCycle();
+    const applications = await this.applicationsRepository.find({
+      where: {
+        //TODO q: I had to change Cycle definition to make year and semester public. Is there a reason it was private?
+        year: currentCycle.year,
+        semester: currentCycle.semester,
+      },
+    });
+
+    const dtos: GetApplicationResponseDTO[] = [];
+
+    applications.forEach((app) =>
+      //TODO q: what is the numApps parameter? I just passed 0 in
```

Comment:
`numApps` refers to the number of times a particular candidate has applied in the past. You can access this variable by changing this to 

```javascript
applications.forEach((app) => {
     const apps = await this.findAll(app.user.id); // all applications associated with the current user
     dtos.push(app.toGetApplicationResponseDTO(apps.length));
});
```
but now that I think about it, do we care about the `numApps` field for this endpoint? Maybe @chromium-52 has a better suggestion for this part.

Reply:
I've updated it now